datafusion/optimizer/src/optimizer.rs

```diff
@@ -244,12 +256,18 @@ impl Optimizer {
             Arc::new(PushDownLimit::new()),
         ];
 
-        Self::with_rules(rules)
+        Self::with_rules(analyzer_rules, rules)
     }
 
     /// Create a new optimizer with the given rules
-    pub fn with_rules(rules: Vec<Arc<dyn OptimizerRule + Send + Sync>>) -> Self {
-        Self { rules }
+    pub fn with_rules(
+        analyzer_rules: Vec<Arc<dyn AnalyzerRule + Send + Sync>>,
+        rules: Vec<Arc<dyn OptimizerRule + Send + Sync>>,
+    ) -> Self {
+        Self {
+            analyzer_rules,
+            rules,
+        }
     }
```

Comment:
I'm unsure if this is a proper place to customize analyzer rules. But from the code structure `Analyzer` is part of `Optimizer`. And only `Optimizer` is exposed to the context. Please let me know what you think @alamb @jackwener